src/Repositories/MovieRepository.cs

```diff
@@ -18,9 +18,9 @@ public IEnumerable<string> GetGenres()
             return Context.Movies.OrderBy(m => m.Genre).Select(m => m.Genre).Distinct();
         }
 
-        IEnumerable<Movie> IMovieRepository.GetLastMovies(int amount)
+        public Movie GetMovieWithComments(int id)
         {
-            return Context.Movies.OrderBy(m => m.ReleaseDate).Take(amount);
+            return Context.Movies.Include(m => m.Comments).Where(m => m.ID == id).FirstOrDefault();
```

Comment:
this works fine, but you are first loading all the comments for all the movies, and then selecting only one.

Reply:
That's right. Replaced with foreach append comment to builder.